rust/bin-shared/src/linux.rs

```diff
@@ -47,3 +39,17 @@ pub fn udp_socket_factory(socket_addr: &SocketAddr) -> io::Result<UdpSocket> {
     setsockopt(&socket, sockopt::Mark, &FIREZONE_MARK)?;
     Ok(socket)
 }
+
+#[cfg(test)]
+mod tests {
+    use super::*;
+    use clap::ValueEnum;
+
+    #[test]
+    fn cli() {
+        assert!(matches!(
+            DnsControlMethod::from_str("etc-resolv-conf", false),
+            Ok(DnsControlMethod::EtcResolvConf)
+        ));
+    }
+}
```

Comment:
I'd prefer more high-level tests around the entire CLI instead of small components like this. Reason being is that we might change how things work under the hood but keep backwards-compatibility with what the user interacts with.

Reply:
Yep I'll get rid of that since it's covered by a bigger test I added later